### PR TITLE
Fix: Upgrade lowlight to 3.1.0 and update demo (CodeBlockLowlight)

### PR DESCRIPTION
**Pull request by RakeshPotnuru**
## Please describe your changes

- Upgraded `lowlight` to `3.1.0`
- Updated demo, example, experiment for `CodeBlockLowlight`.

## How did you accomplish your changes

By reading lowlight docs.

## How have you tested your changes

Yes, tested. The build is successful.

## How can we verify your changes

Read lowlight v3 [Changelog](https://github.com/wooorm/lowlight/releases/tag/3.0.0).

## Checklist

- [x] The changes are not breaking the editor
- [x] Added tests where possible
- [x] Followed the guidelines
- [x] Fixed linting issues

## Related issues

#4456

### CODE REVIEW

**Comment by bdbch** [comment]
@RakeshPotnuru sorry for coming back so late to this. Could you reoslve the merge conflicts and rerun the tests via your commit so we can get this merged?

**Comment by RakeshPotnuru** [comment]
@bdbch, it looks like conflicts are resolved.

File: packages/extension-collaboration/src/collaboration.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import {
 } from 'y-prosemirror'
 import { UndoManager } from 'yjs'
 
+type YSyncOpts = Parameters<typeof ySyncPlugin>[1]
```
Comment (bdbch) [inline packages/extension-collaboration/src/collaboration.ts:12]: Do you know where the changes on this file come from? Seems like they are unrelated to your changes, right?

**Review by bdbch — CHANGES_REQUESTED** [review]
@RakeshPotnuru could you take a look at the changes in the collaboration extension? seems like they were changed unpurposely?

**Comment by RakeshPotnuru** [comment]
@bdbch  I thing it's because multiple PRs got merged into `develop` branch from their respective forks. Should I create a new PR?

image

**Comment by bdbch** [comment]
Yep I think you'd need to rebase your PR to main and keep out the changes from develop.

**Comment by RakeshPotnuru** [comment]
#5036

**Pull request closed without merging**